pythonforandroid/recipe.py

```diff
@@ -993,25 +1019,36 @@ def install_python_package(self, arch, name=None, env=None, is_dir=True):
                     '--install-lib=.',
                     _env=hpenv, *self.setup_extra_args)
 
-            # If asked, also install in the hostpython build dir
-            if self.install_in_hostpython:
-                self.install_hostpython_package(arch)
+            if isfile("setup.py"):
+                shprint(hostpython, 'setup.py', 'install', '-O2',
+                        '--root={}'.format(self.ctx.get_python_install_dir(arch.arch)),
+                        '--install-lib=.',
+                        _env=hpenv, *self.setup_extra_args)
+
+                # If asked, also install in the hostpython build dir
+                if self.install_in_hostpython:
+                    self.install_hostpython_package(arch)
+            else:
+                warning("`PythonRecipe.install_python_package` called without `setup.py` file!")
 
-    def get_hostrecipe_env(self, arch):
+    def get_hostrecipe_env(self, arch=None):
         env = environ.copy()
-        env['PYTHONPATH'] = self.hostpython_site_dir
+        _python_path = self._host_recipe.get_path_to_python()
+        env['PYTHONPATH'] = self._host_recipe.site_dir + ":" + join(
+            _python_path, "Modules") + ":" + glob.glob(join(_python_path, "build", "lib*"))[0]
```

Comment:
In my test this line fails because the first time it is executed the directory is not yet created.
I got this working with:
```python
    def get_hostrecipe_env(self, arch=None):
        env = environ.copy()
        _python_path = self._host_recipe.get_path_to_python()
        libdir = glob.glob(join(_python_path, "build", "lib*"))
        env['PYTHONPATH'] = self._host_recipe.site_dir + ":" + join(
            _python_path, "Modules") + ":" + (libdir[0] if libdir else "")
        return env
```

Reply:
Thanks for testing!